lib/rules/curly.js

```diff
@@ -290,7 +290,7 @@ module.exports = {
                 }
             } else if (multiOrNest) {
                 if (hasBlock && body.body.length === 1 && isOneLiner(body.body[0])) {
-                    expected = false;
+                    expected = !!(body.body[0].leadingComments && body.body[0].leadingComments.length);
```

Comment:
Can we use `sourceCode.getComments(body.body[0]).leading` instead of `body.body[0].leadingComments`. 

We have a few other cases of incorrectly using the `.leadingComments` property in the codebase, but as we are actively moving towards calculating location of comments relative to nodes in ESLint instead of the parser, this will save us having to change this down the road. Thanks!